File: Source/ecs/Source/Library/philipsecommercesdk/src/main/java/com/philips/platform/ecs/model/asset/Assets.java
/* Copyright (c) Koninklijke Philips N.V., 2018
 * All rights are reserved. Reproduction or dissemination
 * in whole or in part is prohibited without the prior written
 * consent of the copyright holder.
 */
package com.philips.platform.ecs.model.asset;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.philips.platform.ecs.util.PRXImageAssetFilter;
import com.philips.platform.ecs.util.PRXImageAssetFilter;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Description :
 * Project : PRX Common Component.
 */


/**
 * The type Assets contains list of assets
 */
public class Assets implements Serializable {

    private static final long serialVersionUID = 3594476307105100868L;
    @SerializedName("asset")
    @Expose
    private List<Asset> asset = new ArrayList<Asset>();

    /**
     * No args constructor for use in serialization
     *
     */
    public Assets() {
    }

    /**
     *
     * @param asset
     */
    public Assets(List<Asset> asset) {
        this.asset = asset;
    }

    /**
     *
     * @return
     * The asset
     */
    public List<Asset> getAsset() {
        return asset;
    }

    public List<Asset> getValidPRXImageAssets(){
        return new PRXImageAssetFilter().getValidPRXAssets(asset);
    }

    /**
     *
     * @param asset
     * The asset
     */
    public void setAsset(List<Asset> asset) {
        this.asset = asset;
    }

}

File: Source/ecs/Source/Library/philipsecommercesdk/src/main/java/com/philips/platform/ecs/microService/model/asset/Assets.kt
/*
 *  Copyright (c) Koninklijke Philips N.V., 2020
 *
 *  * All rights are reserved. Reproduction or dissemination
 *
 *  * in whole or in part is prohibited without the prior written
 *
 *  * consent of the copyright holder.
 *
 *
 */
package com.philips.platform.ecs.microService.model.asset

import android.os.Parcelable
import com.philips.platform.ecs.microService.prx.PRXImageAssetFilter

import kotlinx.android.parcel.Parcelize
import java.io.Serializable
import java.util.*

/**
 * Description :
 * Project : PRX Common Component.
 * Created by naveen@philips.com on 02-Nov-15.
 */
/**
 * The type Assets contains list of assets
 */
@Parcelize
data class Assets(var asset: List<Asset> = ArrayList()) : Parcelable {


    fun getValidPRXImageAssets() : List<Asset>{
      return PRXImageAssetFilter().getValidPRXAssets(asset)
    }

}